### PR TITLE
Pin NPM version and use latest package versions

### PR DESCRIPTION
**Pull request by bion**
- [x] pins NPM version to 8.5.1 which will stop the annoying package-lock.json diffs from constantly getting generated
- [x] upgrades the docker image packages at build time. this improves the security of the image
- [x] adds `bin/dev-bash` script for convenience of getting a bash shell running in a dev container

### CODE REVIEW

**Review by shanemc-goog — APPROVED** [review]

File: bin/dev-bash [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#! /usr/bin/env bash
+
+# DOC: Open a bash shell in the dev environment container.
+
+source bin/lib.sh
+
+bin/pull-image
+
+docker run -it --rm \
```
Comment (goldblatt) [inline bin/dev-bash:9]: i guess a lil nit if we are doing all the inputs on new lines maybe these should be new lined too?

File: Dockerfile [hunk 2]
```diff
@@ -17,7 +17,10 @@ ENV SBT_URL "https://github.com/sbt/sbt/releases/download/v${SBT_VERSION}/sbt-${
 
 RUN set -o pipefail && \
     apk update && \
+    apk add --upgrade apk-tools && \
+    apk upgrade --available && \
     apk add --no-cache --update bash wget npm git openssh && \
+    npm install -g npm@8.5.1 && \
```
Comment (goldblatt) [inline Dockerfile:23]: nit? why not pin to @latest ?

Comment (bion) [reply Dockerfile:23]: on the theory that explicit versioning is better than implicit? i could be convinced otherwise

File: bin/dev-bash (hunk 1 above)
Comment (bion) [reply bin/dev-bash]: done

Comment (bion) [reply bin/dev-bash]: done

File: Dockerfile (hunk 2 above)
Comment (goldblatt) [reply Dockerfile:23]: nah this is all good just curious

**Review by goldblatt — APPROVED** [review]

**Pull request merged**